src/benchmark/trulens/benchmark/benchmark_frameworks/experiments/dataset_preprocessing.py

```diff
@@ -0,0 +1,467 @@
+import ast
+import csv
+import json
+import random
+from typing import List, Tuple
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+from sklearn.metrics import confusion_matrix
+from trulens.feedback import GroundTruthAggregator
+
+
+def generate_summeval_groundedness_golden_set(
+    file_path, max_samples_per_bucket=200
+):
+    """
+    Generate a balanced groundedness golden set from the Summeval dataset.
+
+    Args:
+        file_path (str): Path to the JSON file.
+        max_samples_per_bucket (int): Maximum number of samples per score bucket.
+
+    Yields:
+        dict: A dictionary containing query, expected_response, expected_score, and human_score.
+    """
+
+    def calculate_expected_score(normalized_metrics_lst, weights_lst):
+        """Calculate the expected score using normalized metrics and weights."""
+        assert len(normalized_metrics_lst) == len(weights_lst)
+        return round(
+            sum(
+                normalized_metrics_lst[i] * weights_lst[i]
+                for i in range(len(normalized_metrics_lst))
+            )
+            / sum(weights_lst),
+            2,
+        )
+
+    # Track the number of samples in each bucket
+    buckets = {
+        "low": 0,  # Scores between 0.0 and 0.3
+        "medium": 0,  # Scores between 0.3 and 0.7
+        "high": 0,  # Scores between 0.7 and 1.0
+    }
+
+    with open(file_path, "r", encoding="utf-8") as f:
+        for line in f:
+            # Parse each line as a JSON object
+            try:
+                data = json.loads(line)
+
+                # Ensure the expected keys exist in the JSON
+                try:
+                    row = data
+                    assert (
```

Comment:
Is this method intended only for use with the notebook and summeval? Would it make sense to try to generalize this to support other schemas if we want to incorporate it in the open source package?

Reply:
so for now this method is quite specific to SummEval